src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,128 @@
+                        server,
+                        &mut range_specs,
+                        0,
+                    );
+                    let version = meta.version;
+                    let command = formatdoc!(
+                        "set-option buffer lsp_indent_guides {version} {range_specs}
+                         set-option buffer lsp_indent_guides_timestamp {version}"
+                    );
+                    let command = format!(
+                        "evaluate-commands -buffer {} -- {}",
+                        editor_quote(&meta.buffile),
+                        editor_quote(&command)
+                    );
+                    ctx.exec(meta, command);
+                }
+                _ => {
+                    return;
+                }
+            };
+        },
+    );
+}
+
+fn indent_guides_calc_range_specs(
+    symbols: &[DocumentSymbol],
+    params: &IndentGuidesParams,
+    meta: &EditorMeta,
+    ctx: &Context,
+    server: &ServerSettings,
+    acc: &mut String,
+    level: usize,
+) {
+    for symbol in symbols {
+        let should_skip = level < params.skip;
+
+        let mut filename_path = PathBuf::default();
+        let filename = symbol_filename(meta, symbol, &mut filename_path).to_string();
+        let range = get_kakoune_range(server, filename.as_str(), &symbol.range, ctx).unwrap();
+
+        let children = symbol.children();
+
+        if !should_skip {
+            let is_not_in_child = children
+                .iter()
+                .find(|child| {
+                    let mut filename_path = PathBuf::default();
+                    let filename = symbol_filename(meta, *child, &mut filename_path).to_string();
+                    let range =
+                        get_kakoune_range(server, filename.as_str(), &child.range, ctx).unwrap();
+                    range.start.line <= params.position_line
+                        && params.position_line <= range.end.line
+                })
+                .is_none();
+
+            let is_inside =
+                range.start.line <= params.position_line && params.position_line <= range.end.line;
+            let is_focused = params.highlight_focused && is_not_in_child && is_inside;
+            let face = if is_focused {
+                "IndentGuideFocused"
+            } else {
+                "IndentGuide"
+            };
+
+            let adjusted_level = level - params.skip;
+            let indent_character = params
+                .characters
+                .get(adjusted_level)
+                .or(params.characters.last())
+                .unwrap();
+
+            // We want to ignore putting indentation on signatures and comments,
+            // thus starting from `selection_range`.
+            for line in (symbol.selection_range.end.line + 2)..(range.end.line) {
+                let spec = format!(
+                    "{}.{}+0|{{{}}}{} ",
+                    line, range.start.column, face, indent_character
```

Comment:
`indent_character` should be escaped using `escape_tuple_element()`

Reply:
Done